framework/doc/content/source/userobject/RadialAverage.md

```diff
@@ -0,0 +1,41 @@
+# RadialAverage
+
+!syntax description /UserObjects/RadialAverage
+
+Given a material property and a radius for averaging. The RadialAverage object
+computes the spatial average value of the property over the radius.
+
+## Applications
+This can be used for nonlocal damage models where the damage_index that is used
+for computing the damage stress is average over a certain radius. This can help
+alleviate mesh sensitivity in certain cases.
```

Comment:
Somewhere here, I think it's important to provide a link to the NonlocalDamage/ADNonlocalDamage adapter material and discuss the flow of the local damage model -> RadialAverage -> NonlocalDamage -> Stress calculator